### PR TITLE
Html onchange added in dropdown and js dynamic states and city selection.

### PR DESCRIPTION
**Pull request by Harshwardhanwtagline**

### CODE REVIEW

File: state_city.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }else if(stateDropdown.toLowerCase()=="dehli"){
+        var cities = ["Gokalpuri", "New-Dehli", "Mandoli", "Taj-Pul"]
+    }else if(stateDropdown.toLowerCase()=="up"){
+        var cities = ["Kanpur", "Agra", "Noida", "Saharanpur"]
+    }else if (stateDropdown.toLowerCase()=="chicago"){
+        var cities = ["Lyons", "ForestPark", "RiverForest", "Lincolnwood"]
+    }else if (stateDropdown.toLowerCase()=="texas"){
+        var cities = ["Austin", "SanAntonio", "Dallas", "Houston"]
+    }else if (stateDropdown.toLowerCase()=="lasvegas"){
+        var cities = ["SpringValley", "Boulder", "Paradise", "Henderson"]
```
Comment (himanshu-suthar-tagline) [inline state_city.js]: Create a json for country_state_city like
country_state_city = 
```
{
   'India': 
        {
            'Gujarat': ['Surat', 'Ahemdabad', 'Vadodara'],
            ...
        }
}
```

and then use this json here.

Comment (Harshwardhanwtagline) [reply state_city.js]: Okay Sir

File: state_city.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    for (var index = 0; index < states.length; index++) {
+        var option = document.createElement("option");
+        option.text = states[index].toUpperCase();
+        stateDropdown.add(option);
+      }
+}
+
+function getStateValue(){
+    let countryDropdown = document.getElementById("country").value;
+    let stateDropdown = document.getElementById("states").value.toLowerCase();
```
Comment (himanshu-suthar-tagline) [inline state_city.js]: Here you are using `toLowerCase()` then why you didn't use it for country? Is there specific reason to do this?

File: state_city.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    },
+    "usa": { 
+      "texas": ["Austin", "SanAntonio", "Dallas", "Houston"],
+      "chicago": ["Lyons", "ForestPark", "RiverForest", "Lincolnwood"],
+      "lasVegas": ["SpringValley", "Boulder", "Paradise", "Henderson"]
+    }
+  };
+function getCountryValue(){
+    let countryDropdown = document.getElementById("country").value;
+    var stateDropdown = document.getElementById("states");
```
Comment (himanshu-suthar-tagline) [inline state_city.js]: I think, give variable names in snake case like `state_dropdown` and `country_dropdown`. This way we can better distinguish between function name and variable name.

File: state_city.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+
+const data = {
```
Comment (himanshu-suthar-tagline) [inline state_city.js]: Give constant name in capital like `DATA`. Intead of `DATA` give some meaningful name like `COUNTRY_STATE_CITY_DD`

File: state_city.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    if (data[countryDropdown].hasOwnProperty(stateDropdown)){
+        var cities = data[countryDropdown][stateDropdown];
+    };
+
+    for (var index = 0; index < cities.length; index++) {
+        var option = document.createElement("option");
+        option.text = cities[index].toUpperCase();
+        cityDropdown.add(option);
+      }
```
Comment (himanshu-suthar-tagline) [inline state_city.js]: Can you reuse the lines 21-25 as you can see here and there both lines are repeating?

File: state_city.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    state_dropdown.innerHTML = ""
+    if (COUNTRY_STATE_CITY_DD.hasOwnProperty(country_dropdown)){
+        var states = Object.keys(COUNTRY_STATE_CITY_DD[country_dropdown]);
+    }
+    state_dropdown = addOption(state_dropdown, states);
+}
+
+function getStateValue(){
+    let country_dropdown = document.getElementById("country").value;
+    let state_dropdown = document.getElementById("states").value.toLowerCase();
```
Comment (himanshu-suthar-tagline) [inline state_city.js:36]: Here you are using the `toLowerCase()` and not on the above part. Why?

File: index.html [hunk 7]
```diff
@@ -0,0 +1,89 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.1/jquery.min.js"></script>
+    <script src="state_city.js"></script>
```
Comment (himanshu-suthar-tagline) [inline index.html]: Import this script in the body tag at the end of body tag.

File: state_city.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  for (var index = 0; index < array.length; index++) {
+    var option = document.createElement("option");
+    option.text = array[index].toUpperCase();
+    dropdown.add(option);
+  }
+  return dropdown;
+}
+function getCountryValue(){
+    let country_dropdown = document.getElementById("country").value.toLowerCase();
+    var state_dropdown = document.getElementById("states");
```
Comment (himanshu-suthar-tagline) [inline state_city.js]: Why you are using `var` here when you have used `let` in below at line 35. Also, you are not accessing the value here after the `getElementById("states")` whereas in below you are using `.value` after this. Please take time to review your code before you asked review from side. So you can find your mistakes on your own.

**Pull request merged**